Python/specialize.c

```diff
@@ -1764,6 +1772,16 @@ _Py_Specialize_CompareOp(PyObject *lhs, PyObject *rhs,
         when_to_jump_mask = (1 | 2 | 4) & ~when_to_jump_mask;
     }
     if (Py_TYPE(lhs) != Py_TYPE(rhs)) {
+#ifdef Py_STATS
+        if (PyFloat_CheckExact(lhs) && PyLong_CheckExact(rhs)) {
+            SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_FLOAT_LONG);
+            goto failure;
+        }
+        if (PyLong_CheckExact(lhs) && PyFloat_CheckExact(rhs)) {
+            SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_LONG_FLOAT);
+            goto failure;
+        }
+#endif
         SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_DIFFERENT_TYPES);
```

Comment:
Could you rewrite this as `SPECIALIZATION_FAIL(COMPARE_OP, compare_op_fail_kind(lhs, rhs));` following the pattern used in the other specialization functions?